File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
  def isValid(self, s: str) -> bool:
    dict = { ']': '[', ')': '(', '}': '{' }
    stack = []
    for chr in s:
      if chr in dict:
        if len(stack) == 0 or stack.pop() != dict[chr]:
          return False
      else:
        stack.append(chr)
    return len(stack) == 0
        

File: 0020-valid-parentheses/0020-valid-parentheses.php
class Solution {

  /**
    * @param String $s
    * @return Boolean
    */
  function isValid($s) {
    $pairs = [']' => '[', ')' => '(', '}' => '{'];
    $stack = [];
    foreach (str_split($s) as $ch) {
      if ($pairs[$ch]) {
        if (empty($stack) || array_pop($stack) !== $pairs[$ch]) return false;
      } else {
        $stack[] = $ch;
      }
    }
    return empty($stack);
  }
}